sycl/include/CL/sycl/handler.hpp

```diff
@@ -769,15 +774,16 @@ class __SYCL_EXPORT handler {
         (KI::callsThisItem());
 
     // Perform range rounding if rounding-up is enabled
+    // and there are sufficient work-items to need rounding
     // and the user-specified range is not a multiple of a "good" value.
-    if (!DisableRounding && NumWorkItems[0] % GoodLocalSizeX != 0) {
+    if (!DisableRounding && (NumWorkItems[0] > MinRangeX) &&
```

Comment:
```suggestion
    if (!DisableRounding && (NumWorkItems[0] >= MinRangeX) &&
```
According to the description is should be more or equal.